components/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState, useRef } from "react";
+import Icon from "../Icon/Icon";
+import "./ProgressBar.scss";
+
+type Props = {
+  value: number;
+  onChange: (newProgress: number) => void;
+};
+
+const ProgressBar: React.FC<Props> = ({ value, onChange }) => {
+  const [isDragging, setIsDragging] = useState(false);
+  const [dragStartX, setDragStartX] = useState(0);
+  const [dragStartValue, setDragStartValue] = useState(value);
+  const progressBarRef = useRef<HTMLDivElement>(null);
+
+  const handleIconMouseDown = (event: React.MouseEvent<HTMLDivElement>) => {
+    event.stopPropagation();
+    setIsDragging(true);
+    setDragStartX(event.clientX);
+    setDragStartValue(value);
+  };
+
+  const calculatePinPosition = (event: React.MouseEvent<HTMLDivElement>) => {
+    const progressBarRect = progressBarRef.current?.getBoundingClientRect();
+    const progressBarWidth = progressBarRect?.width || 0;
+    const offsetX = event.clientX - (progressBarRect?.left || 0);
+    const progressPercentage = offsetX / progressBarWidth;
+
+    const clampedPercentage = Math.max(0, Math.min(1, progressPercentage));
+    const newValue = Math.round(clampedPercentage * 100);
```

Comment:
NIT: It's just about readability :) 
```suggestion
    const { left, width } = progressBarRef.current?.getBoundingClientRect() ?? {};

    const offsetX = event.clientX - left;
    const progressPercentage = offsetX / width;

    const clampedPercentage = Math.max(0, Math.min(1, progressPercentage));
    const newValue = Math.round(clampedPercentage * 100);
```